tests/compile/fsm-guard.fil

```diff
@@ -0,0 +1,54 @@
+// Tests the guards of an fsm.
+
+comp test2[#DELAY]<G: #DELAY>(
+    in: [G, G+#DELAY] 32,
+) -> (
+    out[#DELAY]: for<#i> [G+#i, G+#i+1] 32
+) where #DELAY > 0 {
+    for #ii in 0..#DELAY {
+        out{#ii} = in;
+    }
+}
+
+comp test[#DELAY, #NUM]<G: #DELAY>(
```

Comment:
Any chance we could give this component a slightly more informative name? If not, maybe a comment explaining what it is doing within the test?